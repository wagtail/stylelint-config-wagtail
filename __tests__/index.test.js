const fs = require('fs');
const stylelint = require('stylelint');

const config = require('..');

describe('flags no warnings with valid css', () => {
  const code = fs.readFileSync('./__tests__/scss-valid.scss', 'utf-8');

  let result;

  beforeEach(() => {
    result = stylelint.lint({ code, config });
  });

  it('did not error', (done) => {
    expect.assertions(1);

    result.then((data) => {
      expect(data.errored).toBeFalsy();

      done();
    });
  });

  it('flags no warnings', (done) => {
    expect.assertions(1);

    result.then((data) => {
      expect(data.results[0].warnings).toHaveLength(0);

      done();
    });
  });
});

describe('flags warnings with invalid css', () => {
  const code = fs.readFileSync('./__tests__/scss-invalid.scss', 'utf-8');

  let result;

  beforeEach(() => {
    result = stylelint.lint({ code, config });
  });

  it('did error', (done) => {
    expect.assertions(2);

    result.then((data) => {
      expect(data.errored).toBeTruthy();

      // output unique error names to a snapshot
      expect(
        data.results[0].warnings
          .map(({ rule }) => rule)
          .reduce(
            (rulesWithCount, rule) => ({
              ...rulesWithCount,
              [rule]: (rulesWithCount[rule] || 0) + 1,
            }),
            {},
          ),
      ).toMatchSnapshot();

      done();
    });
  });
});
