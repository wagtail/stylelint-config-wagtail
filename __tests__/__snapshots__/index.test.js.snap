// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flags warnings with invalid css did error 1`] = `
{
  "block-no-empty": 1,
  "color-named": 1,
  "comment-no-empty": 1,
  "declaration-block-no-duplicate-properties": 1,
  "declaration-property-value-allowed-list": 2,
  "declaration-property-value-disallowed-list": 3,
  "length-zero-no-unit": 1,
  "max-nesting-depth": 1,
  "order/order": 1,
  "property-disallowed-list": 3,
  "rule-empty-line-before": 8,
  "scale-unlimited/declaration-strict-value": 9,
  "scss/at-rule-no-unknown": 1,
  "scss/comment-no-empty": 1,
  "scss/selector-no-union-class-name": 1,
  "selector-attribute-name-disallowed-list": 1,
  "selector-class-pattern": 2,
  "selector-max-combinators": 2,
  "selector-max-id": 1,
  "selector-max-specificity": 5,
  "unit-no-unknown": 1,
}
`;
